src/analysis/normed_space/normed_group_hom.lean

```diff
@@ -0,0 +1,461 @@
+    the operator norm. -/
+instance to_normed_group : normed_group (normed_group_hom V₁ V₂) :=
+normed_group.of_core _ ⟨op_norm_zero_iff, op_norm_add_le, op_norm_neg⟩
+
+/-- Coercion of a `normed_group_hom` is an `add_monoid_hom`. Similar to `add_monoid_hom.coe_fn` -/
+@[simps]
+def coe_fn_add_hom : normed_group_hom V₁ V₂ →+ (V₁ → V₂) :=
+{ to_fun := coe_fn, map_zero' := coe_zero, map_add' := coe_add}
+
+@[simp] lemma coe_sum {ι : Type*} (s : finset ι) (f : ι → normed_group_hom V₁ V₂) :
+  ⇑(∑ i in s, f i) = ∑ i in s, (f i) :=
+(coe_fn_add_hom : _ →+ (V₁ → V₂)).map_sum f s
+
+lemma sum_apply {ι : Type*} (s : finset ι) (f : ι → normed_group_hom V₁ V₂) (v : V₁) :
+  (∑ i in s, f i) v = ∑ i in s, (f i v) :=
+by simp only [coe_sum, finset.sum_apply]
+
+/-! ### Composition of normed group homs -/
+
+/-- The composition of continuous normed group homs. -/
+@[simps]
+protected def comp (g : normed_group_hom V₂ V₃) (f : normed_group_hom V₁ V₂) :
+  normed_group_hom V₁ V₃ :=
+(g.to_add_monoid_hom.comp f.to_add_monoid_hom).mk_continuous (∥g∥ * ∥f∥) $ λ v, calc
+∥g (f v)∥ ≤ ∥g∥ * ∥f v∥ : le_op_norm _ _
+... ≤ ∥g∥ * (∥f∥ * ∥v∥) : mul_le_mul_of_nonneg_left (le_op_norm _ _) (op_norm_nonneg _)
+... = ∥g∥ * ∥f∥ * ∥v∥   : by rw mul_assoc
+
+lemma norm_comp_le (g : normed_group_hom V₂ V₃) (f : normed_group_hom V₁ V₂) :
+  ∥g.comp f∥ ≤ ∥g∥ * ∥f∥ :=
+mk_continuous_norm_le _ (mul_nonneg (op_norm_nonneg _) (op_norm_nonneg _)) _
+
+/-- Composition of normed groups hom as an additive group morphism. -/
+def comp_hom : (normed_group_hom V₂ V₃) →+ (normed_group_hom V₁ V₂) →+ (normed_group_hom V₁ V₃) :=
+add_monoid_hom.mk' (λ g, add_monoid_hom.mk' (λ f, g.comp f)
+  (by { intros, ext, exact g.map_add _ _ }))
+  (by { intros, ext, simp only [comp_apply, pi.add_apply, function.comp_app,
+                                add_monoid_hom.add_apply, add_monoid_hom.coe_mk', coe_add] })
+
+@[simp] lemma comp_zero (f : normed_group_hom V₂ V₃) : f.comp (0 : normed_group_hom V₁ V₂) = 0 :=
+by { ext, exact f.map_zero }
+
+@[simp] lemma zero_comp (f : normed_group_hom V₁ V₂) : (0 : normed_group_hom V₂ V₃).comp f = 0 :=
+by { ext, refl }
+
+end normed_group_hom
+
+namespace normed_group_hom
+
+/-!### Kernel -/
+section kernels
+
+variables {V V₁ V₂ V₃ : Type*}
+variables [normed_group V] [normed_group V₁] [normed_group V₂] [normed_group V₃]
+variables (f : normed_group_hom V₁ V₂) (g : normed_group_hom V₂ V₃)
+
+/-- The kernel of a bounded group homomorphism. Naturally endowed with a `normed_group` instance. -/
+def ker : add_subgroup V₁ := f.to_add_monoid_hom.ker
+
+lemma mem_ker (v : V₁) : v ∈ f.ker ↔ f v = 0 :=
+by { erw f.to_add_monoid_hom.mem_ker, refl }
+
+/-- The inclusion of the kernel, as bounded group homomorphism. -/
+@[simps] def ker.incl : normed_group_hom f.ker V₁ :=
+{ to_fun := (coe : f.ker → V₁),
+  map_add' := λ v w, add_subgroup.coe_add _ _ _,
+  bound' := ⟨1, λ v, by { rw [one_mul], refl }⟩ }
```

Comment:
I don't see anything specific to the kernel here, this inclusion should be defined for any subgroup.